projects/distributed-tracing/src/shared/dashboard/widgets/table/table-widget.model.ts

```diff
@@ -172,8 +172,14 @@ export class TableWidgetModel {
     return this.api.getData<TableDataSource<TableRow>>();
   }
 
-  public getColumns(): SpecificationBackedTableColumnDef[] {
-    return this.columns.map(column => column.asTableColumnDef());
+  public getColumns(mode?: TableMode): SpecificationBackedTableColumnDef[] {
+    return this.columns
+      .filter(column => this.filterFlattenedTreeColumns(column, mode ?? this.mode))
+      .map(column => column.asTableColumnDef());
+  }
+
+  private filterFlattenedTreeColumns(column: TableWidgetColumnModel, mode: TableMode): boolean {
+    return !column.flattenedOnly || (column.flattenedOnly && this.modeOptions.length > 0 && mode === TableMode.Flat);
```

Comment:
Why the length check? Shouldn't a flattened only col always show up if mode is flat?

Reply:
may be moot given the other comments, but If I had a flat table, I'd expect flattenedOnly columns to show up regardless  of having a mode toggle or not.